project/src/pages/main/main.tsx

```diff
@@ -1,68 +1,82 @@
+import { Link } from 'react-router-dom';
 import CardComponent from '../../components/card/card';
-import { HeaderMenu } from '../../components/header-menu/header-menu';
 import { Header } from '../../components/header/header';
 import MapComponent from '../../components/map/map-component';
-import { City, Locations, Offers } from '../../types/offer';
+import { Cities, SortOffers } from '../../const';
+import { Offers } from '../../types/offer';
+import { getOffersByCity } from '../../store/action';
+import { useAppDispatch, useAppSelector } from '../../hooks/store';
+import SortList from '../../components/sort-list/sort-list';
+import { useEffect, useState } from 'react';
+import MainScreenEmpty from '../main-empty/main-empty';
 
-type MainScreenProps = {
-  placesAmount: number;
+type LocationTabItemProps = {
+  isActive: boolean;
+  city: Cities;
+  changeCurrentLocation: (e: React.MouseEvent<HTMLAnchorElement>, city: Cities) => void;
+}
+
+type AllOffersProps = {
   offers: Offers;
-  city: City;
-  locations: Locations;
+  setActiveOffer: (id: number | null) => void;
 }
 
-function MainScreen(props: MainScreenProps): JSX.Element {
-  const { placesAmount, offers, city, locations } = props;
+const getCityOffers = (cityName: string, items: Offers) => items.filter((e) => e.city.name === cityName);
+
+function MainScreen(): JSX.Element {
+  const [currentOffers, setCurrentOffers] = useState<Offers>([]);
+  const [activeOffer, setActiveOffer] = useState<null | number>(null);
+  const dispatch = useAppDispatch();
+  const selectedCity = useAppSelector((state) => state.city);
+  const offers = useAppSelector((state) => state.offers);
+  const offersAll = useAppSelector((state) => state.offers.filter((offer) => offer.city.name === selectedCity));
+  const selectedSortItem = useAppSelector((state) => state.sortOption);
+  const sortOffers = SortOffers(offersAll, selectedSortItem);
+
+  const handleChangeCity = (e: React.MouseEvent<HTMLAnchorElement>, city: Cities) => {
+    e.preventDefault();
+    dispatch(getOffersByCity(city));
+  };
 
-  const styleProp = { height: '682px' };
+  useEffect(() => {
+    const cityOffers = getCityOffers(selectedCity, offers);
+    // cityOffers = SortOffers(cityOffers, currentSortType);
 
+    setCurrentOffers(cityOffers);
+
+  }, [selectedCity, offers]);
+
+  const offersEmpty = currentOffers.length < 1;
   return (
```

Comment:
я бы назвал noOffers
```suggestion
  const noOffers = currentOffers.length < 1;
  return (
```